ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/Conversions.scala

```diff
@@ -70,11 +70,14 @@ private[state] object Conversions {
   }
 
   def commandDedupKey(subInfo: DamlSubmitterInfo): DamlStateKey = {
+    val sortedUniqueSubmitters =
+      if (subInfo.getSubmittersCount == 1) subInfo.getSubmittersList
+      else subInfo.getSubmittersList.asScala.distinct.sorted.asJava
```

Comment:
Instead of going back-and-forth between Scala and Java views this could be simplified using `Collections.sort()`:
```suggestion
      Collections.sort(subInfo.getSubmittersList)
```
If you absolutely want to avoid copying a single submitter then branching on the submitter count may still make sense.

Reply:
> Collections.sort(subInfo.getSubmittersList)

I thought about this, but I do not like performing a mutable operation on a method argument.

> branching on the submitter count may still make sense

Multi-party submissions are expected to be an admin operation used for testing or migrating ledgers. During regular ledger operations, all submissions are expected to have exactly one submitter. That's why I special cased it.

Reply:
Thanks for the clarification.